src/Middleware/HttpLogging/src/MediaTypeOptions.cs

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.HttpLogging
+{
+    /// <summary>
+    /// Options for HttpLogging to configure which encoding to use for each media type.
+    /// </summary>
+    public sealed class MediaTypeOptions
+    {
+        private readonly List<MediaTypeState> _mediaTypeStates = new();
+
+        internal MediaTypeOptions()
+        {
+        }
+
+        internal List<MediaTypeState> MediaTypeStates => _mediaTypeStates;
+
+        internal static MediaTypeOptions BuildDefaultMediaTypeOptions()
+        {
+            var options = new MediaTypeOptions();
+            options.AddText("application/json", Encoding.UTF8);
+            options.AddText("application/*+json", Encoding.UTF8);
+            options.AddText("application/xml", Encoding.UTF8);
+            options.AddText("application/*+xml", Encoding.UTF8);
+            options.AddText("text/*", Encoding.UTF8);
+
+            return options;
+        }
+
+        /// <summary>
+        /// Adds a <see cref="MediaTypeHeaderValue"/> to be used for logging as text.
+        /// </summary>
+        /// <param name="mediaType">The MediaType to add.</param>
+        public void AddText(MediaTypeHeaderValue mediaType)
```

Comment:
I don't think we should expose this as `MediaTypeHeaderValue`, lets expose it as a string.

Reply:
I'm against this, why not expose an API that accepts the public representation of the thing we are checking?

Reply:
I have a personal vendatta against this type and I'm fine using it internally but not in public API. I'd like these typed headers to die and eventually use the ones in .NET.